packages/vite/src/node/server/index.ts

```diff
@@ -396,16 +401,32 @@ export async function createServer(
       }
     },
     async listen(port?: number, isRestart?: boolean) {
-      await startServer(server, port, isRestart)
+      await startServer(server, port)
       if (httpServer) {
         server.resolvedUrls = await resolveServerUrls(
           httpServer,
           config.server,
           config,
         )
       }
+
+      if (!isRestart && config.server.open) server.openDevBrowser()
       return server
     },
+    openDevBrowser() {
+      const options = server.config.server
+      const url = server.resolvedUrls?.local[0]
+      if (url) {
+        const path =
+          typeof options.open === 'string'
+            ? new URL(options.open, url).href
+            : url
+
+        openBrowser(path, true, server.config.logger)
+      } else {
+        server.config.logger.warn('No URL available to open in browser')
+      }
+    },
```

Comment:
I think this method should check if `server.config.open` is truthy first before opening the browser. So we prevent opening it if the user didn't set the option correctly.

Reply:
This method will be used when pressing the shortcut key. The open check should be placed outside. This is a general method

Reply:
Can't open without setting the shortcut key?

Reply:
Oh ya you're right. Let's keep it as is then.